src/azure-cli/setup.py

```diff
@@ -59,68 +59,68 @@
     'azure-cli-nspkg~=3.0,>=3.0.3',
     'azure-cli-telemetry>=1.0.2,<2.0',
     'azure-cosmos~=3.0,>=3.0.2',
-    'azure-datalake-store~=0.0.45',
-    'azure-functions-devops-build~=0.0.22',
-    'azure-graphrbac~=0.60.0',
-    'azure-keyvault~=1.1',
+    'azure-datalake-store=0.0.45',
+    'azure-functions-devops-build=0.0.22',
+    'azure-graphrbac=0.60.0',
+    'azure-keyvault=1.1',
     'azure-mgmt-advisor>=2.0.1,<3.0.0',
-    'azure-mgmt-applicationinsights~=0.1.1',
-    'azure-mgmt-authorization~=0.50.0',
-    'azure-mgmt-batch~=6.0',
-    'azure-mgmt-batchai~=2.0',
-    'azure-mgmt-billing~=0.2',
-    'azure-mgmt-botservice~=0.2.0',
-    'azure-mgmt-cdn~=3.1',
-    'azure-mgmt-cognitiveservices~=3.0',
-    'azure-mgmt-compute~=5.0',
-    'azure-mgmt-consumption~=2.0',
-    'azure-mgmt-containerinstance~=1.4',
-    'azure-mgmt-containerregistry~=2.8',
-    'azure-mgmt-containerservice~=5.2',
-    'azure-mgmt-cosmosdb~=0.5.2',
-    'azure-mgmt-datalake-analytics~=0.2.1',
-    'azure-mgmt-datalake-store~=0.5.0',
-    'azure-mgmt-datamigration~=0.1.0',
-    'azure-mgmt-deploymentmanager~=0.1.0',
-    'azure-mgmt-devtestlabs~=2.2',
-    'azure-mgmt-dns~=2.1',
-    'azure-mgmt-eventgrid~=2.2',
-    'azure-mgmt-eventhub~=2.6',
-    'azure-mgmt-hdinsight~=1.1.0',
-    'azure-mgmt-imagebuilder~=0.2.1',
-    'azure-mgmt-iotcentral~=1.0',
-    'azure-mgmt-iothub~=0.8.2',
-    'azure-mgmt-iothubprovisioningservices~=0.2.0',
-    'azure-mgmt-keyvault~=1.1',
-    'azure-mgmt-kusto~=0.3.0',
-    'azure-mgmt-loganalytics~=0.2',
-    'azure-mgmt-managementgroups~=0.1',
-    'azure-mgmt-maps~=0.1.0',
-    'azure-mgmt-marketplaceordering~=0.1',
-    'azure-mgmt-media~=1.1,>=1.1.1',
-    'azure-mgmt-monitor~=0.5.2',
-    'azure-mgmt-msi~=0.2',
-    'azure-mgmt-network~=3.0',
-    'azure-mgmt-policyinsights~=0.3.1',
-    'azure-mgmt-privatedns~=0.1.0',
-    'azure-mgmt-rdbms~=1.8',
-    'azure-mgmt-recoveryservices~=0.1.1',
-    'azure-mgmt-recoveryservicesbackup~=0.1.2',
-    'azure-mgmt-redis~=6.0',
-    'azure-mgmt-relay~=0.1.0',
-    'azure-mgmt-reservations~=0.3.1',
```

Comment:
This ~ caused and could easily cause in the future for an SDK release to suddenly break. In this case, `azure-mgmt-reservations 0.3.2` bumps the API version, causing all test recordings to fail.

Reply:
Got it. The typical solution to that is to check-in a comprehensive requirements.txt. Shouldn't be too complicated for me to do, we can talk more offline.